pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -941,13 +941,15 @@ ClientConnection::newConsumerStats(uint64_t consumerId, uint64_t requestId) {
 void ClientConnection::newTopicLookup(const std::string& destinationName, bool authoritative,
                                       const uint64_t requestId,
                                       LookupDataResultPromisePtr promise) {
+    Lock lock(mutex_);
```

Comment:
but that's fine..right? as far as specific command get back `cmd.mutable_X()` is thread-safe, there should not be any issue. 
in case of lookup: if serializing lookup data  is thread-safe, it will not affect any lookup or other cmd serialization.
right?